scipy/fft/_backend.py

```diff
@@ -0,0 +1,169 @@
+import scipy._lib.uarray as ua
+import scipy.fftpack as _fftpack
+from . import _pocketfft
+
+
+class _ScipyBackend:
+    """
+    The default backend for fft calculations
+
+    Notes
+    -----
+    We use the domain `numpy.scipy` because in the future `uarray` will treat
+    the domain as a heirarchy where `numpy.scipy` may call into a `numpy`
+    backend.
+
+    """
+    __ua_domain__ = "numpy.scipy.fft"
+
+    @staticmethod
+    def __ua_function__(method, args, kwargs):
+        fn = (getattr(_pocketfft, method.__name__, None)
+              or getattr(_fftpack, method.__name__, None))
+
+        if fn is None:
+            return NotImplemented
+        return fn(*args, **kwargs)
+
+
+_named_backends = {
+    'scipy': _ScipyBackend,
+}
+
+def _backend_from_arg(backend):
+    """Maps strings to known backends and validates the backend"""
+
+    if isinstance(backend, str):
+        try:
+            backend = _named_backends[backend]
+        except KeyError:
+            raise ValueError('Unknown backend {}'.format(backend))
+
+    if backend.__ua_domain__ != 'numpy.scipy.fft':
+        raise ValueError('Backend does not implement "numpy.scipy.fft"')
+
+    return backend
+
+
+def set_global_backend(backend):
+    """Sets the global fft backend
+
+    The global backend has higher priority than registered backends, but lower
+    priority than context-specific backends set with `set_backend`.
+
+    Parameters
+    ----------
+    backend: {object, 'scipy'}
+
+        The backend to use.
+        Can either be a ``str`` containing the name of a known backend
+        {'scipy'}, or an object that implements the uarray protocol.
+
+    Raises
+    ------
+    ValueError: If the backend does not implement ``numpy.scipy.fft``
+
+    Notes
+    -----
+    This will overwrite the previously set global backend, which by default is
+    the SciPy implementation.
+
```

Comment:
```suggestion
```

Reply:
I've been going by the [numpydoc guide](https://numpydoc.readthedocs.io/en/latest/format.html) which seems to always have a blank line at the end. Does SciPy do this differently?

Reply:
I'm used to there not being a blank line at the end but you're right that it's fine.

Reply:
Restructuredtext requires it, Numpydoc doesn't, neither does Napoleon. Some people just prefer ReST everywhere, some just don't make docstrings too long. It's a matter of choice. 🤷‍♂